app/app-controllers/searchBarController.js

```diff
@@ -20,15 +23,13 @@ function(
         this.isVisible = true;
 
         this._registerEvents();
-
-        this.viewPlugin.navigate(this.searchConfig.uiSource);
-        this.viewPlugin.disableScrolling();
     };
 
     SearchBarController.init = function(layoutPromise, searchConfig) {
-        return Promise.join(layoutPromise, WebViewPlugin.init(),
-            function(layout, webView) {
-                return new SearchBarController(webView, layout, searchConfig);
+        return Promise.join(layoutPromise, SearchBarPlugin.init(), AnchoredLayoutPlugin.init(),
+            function(layout, searchBarPlugin, internalLayout) {
+                internalLayout.addTopView(searchBarPlugin);
+                return new SearchBarController(searchBarPlugin, layout, internalLayout, searchConfig);
```

Comment:
While you are editing this, can you make the promise resolution stuff be the same order as the constructor call? It just makes it a bit easier to follow.